x-pack/plugins/security_solution/public/detection_engine/rule_management/api/api.ts

```diff
@@ -213,72 +213,68 @@ export interface BulkActionResponse {
   };
 }
 
-export interface BulkActionProps {
-  action: Exclude<BulkAction, BulkAction.export>;
-  query?: string;
-  ids?: string[];
-  edit?: BulkActionEditPayload[];
-  isDryRun?: boolean;
-}
+export type QueryOrIds = { query: string; ids?: undefined } | { query?: undefined; ids: string[] };
+type PlainBulkActionDescriptor = {
+  type: Exclude<BulkAction, BulkAction.edit | BulkAction.export>;
+} & QueryOrIds;
+type EditBulkActionDescriptor = {
+  type: BulkAction.edit;
+  editPayload: BulkActionEditPayload[];
+} & QueryOrIds;
+export type BulkActionDescriptor = PlainBulkActionDescriptor | EditBulkActionDescriptor;
 
 /**
  * Perform bulk action with rules selected by a filter query
  *
+ * @param type bulk action to perform
  * @param query filter query to select rules to perform bulk action with
- * @param ids string[] rule ids to select rules to perform bulk action with
+ * @param ids rule ids to select rules to perform bulk action with
  * @param edit BulkEditActionPayload edit action payload
- * @param action bulk action to perform
- * @param isDryRun enables dry run mode for bulk actions
+ * @param dryRun enables dry run mode for bulk actions
  *
  * @throws An error if response is not OK
  */
-export const performBulkAction = async ({
-  action,
-  query,
-  edit,
-  ids,
-  isDryRun,
-}: BulkActionProps): Promise<BulkActionResponse> =>
-  KibanaServices.get().http.fetch<BulkActionResponse>(DETECTION_ENGINE_RULES_BULK_ACTION, {
+export async function performBulkAction(
+  bulkActionDescriptor: BulkActionDescriptor,
+  dryRun?: boolean
```

Comment:
Let's use an object for method arguments for consistency with other methods in this file. It also makes it much more understandable for other developers to see what is being passed to the function.

Compare:

```ts
// What does this "true" mean? It is not possible to say without looking into the function's implementation
performBulkAction(bulkActionDescriptor, true)

// It's now clear that "true" enables the dry run mode
performBulkAction({ bulkActionDescriptor, dryRun: true })
```